pc-air/pc-air.wdl

```diff
@@ -0,0 +1,388 @@
+        File pcair_output_unrelated = if defined(out_prefix) then "~{out_prefix}_pca_unrel.RData" else "pca_unrel.RData"
+        File config_file            = "pca_byrel.config"
+    }
+
+}
+
+# [3] pca_plots -- plots the PCs
+task pca_plots {
+    input {
+        File pca_file
+
+        # optional
+        File? phenotype_file
+
+        Int n_pairs = 6
+        String? group = "NA"
+        String? out_prefix
+        
+        # runtime attributes
+        Int addldisk = 5
+        Int cpu = 2
+        Int memory = 4
+        Int preempt = 3
+    }
+
+    # Estimate disk size required
+    Int pca_size        = ceil(size(pca_file, "GB"))
+    Int final_disk_dize = pca_size + addldisk
+
+    # Are optional files definec?
+    Boolean defPhenotypeFile     = defined(phenotype_file)
+    
+
+    command <<<
+        set -eux -o pipefail
+
+        echo "Generating config file"
+        # noquotes = argument has no quotes in the config file to match CWL config EXACTLY
+        python << CODE
+        import os
+
+        f = open("pca_plots.config", "a")
+
+        f.write('pca_file "~{pca_file}"\n')
+        
+        if ~{n_pairs} != 6:
+            f.write("n_pairs ~{n_pairs}\n")
+
+        if "~{out_prefix}" != "":
+                f.write('out_file_scree "~{out_prefix}_pca_scree.pdf"\n')
+                f.write('out_file_pc12 "~{out_prefix}_pca_pc12.pdf"\n')
+                f.write('out_file_parcoord "~{out_prefix}_pca_parcoord.pdf"\n')
+                f.write('out_file_pairs "~{out_prefix}_pca_pairs.png"\n')
+
+        if "~{group}" != "NA":
+            f.write("group ~{group}\n")
+
+        if "~{defPhenotypeFile}" == "true":
+            f.write('phenotype_file "~{phenotype_file}"\n') 
+
+        f.close()
+
+        CODE
+
+        echo "Calling R script pca_plots.R"
+        Rscript //usr/local/analysis_pipeline/R/pca_plots.R pca_plots.config
+    >>>
+
+    runtime {
+        cpu: cpu
+        docker: "uwgac/topmed-master@sha256:0bb7f98d6b9182d4e4a6b82c98c04a244d766707875ddfd8a48005a9f5c5481e"
+        disks: "local-disk " + final_disk_dize + " HDD"
+        memory: "${memory} GB"
+        preemptibles: "${preempt}"
+    }
+    output {
+        Array[File] pca_plots  = glob("*.p*")
+        File config_file  = "pca_plots.config"
+    }
+
+}
+
+# [4] variant_id_from_gds -- (optional) pull variant IDs from GDS file
+task variant_id_from_gds {
+    input {
+        File gds_file
+        String? out_file
+        
+        # runtime attributes
+        Int addldisk = 5
+        Int cpu = 2
+        Int memory = 4
+        Int preempt = 3
+    }
+
+    # Estimate disk size required
+    Int gds_size        = ceil(size(gds_file, "GB"))
+    Int final_disk_dize = gds_size + addldisk
+    
+    command <<<
+        set -eux -o pipefail
+
+        echo "Generating variant_id_from_gds.R w/ HereDoc"
+        
+        cat << 'EOF' > variant_id_from_gds.R
```

Comment:
Okay, so like, this in the CWL and therefore should be here, but I have to ask...

Why? 🤔

Reply:
You mean the script? It is just pulling the variant IDs. They are used as a variant_include_file in the correlation wf.